### ISSUE TITLE
Room for improvement in the release profile?

### ISSUE THREAD

**Issue by CramBL** [emilk/eframe_template#163]
https://github.com/emilk/eframe_template/blob/ac47eafb17cef09c4b3cb0ba4f523b68367bdd57/Cargo.toml#L41

I might be missing something but why not prefer a profile such as the following?
```toml
[profile.release]
opt-level = 3
debug = false
panic = "unwind"
lto = true
codegen-units = 1
strip = true
```

Users have to download a slightly larger wasm **ONCE** while better performance is continuously valuable, or what am I missing here?

**Comment by emilk** [comment]
You suggestion leads to slower compile times, larger wasm, and lack of debug symbols. That may be a good trade-off for some people, for others not, which is probably why those aren't the defaults set by cargo/rustc.

In the end, this is only a template - users can change it as they see fit.

**Comment by CramBL** [comment]
Sorry if I came across as rude, I don't know much about WASM so I was mostly looking to test my assumptions before potentially ruining the build configuration in my project.

For what it's worth the proposed build configuration **reduces** wasm size for this project (and my own) for the template it goes from `2848213` to  `2795274` with `trunk build --release`. It might simply be because of the stripped debug symbols but it doesn't bloat wasm.

Anyways, thanks for the reply and thank you for this awesome project, have a nice day! :star:

**Comment by berkus** [comment]
@CramBL actually, try `opt-level = "s"` or `opt-level = "z"` it might improve the size quite a bit even for wasm.

I'd try out this for release:

```
[profile.release]
opt-level = "s"
lto = true
codegen-units = 1
panic = "abort"
strip = "symbols"
```

**Comment by CramBL** [comment]
> @CramBL actually, try `opt-level = "s"` or `opt-level = "z"` it might improve the size quite a bit even for wasm.
> 
> I'd try out this for release:
> 
> ```
> [profile.release]
> opt-level = "s"
> lto = true
> codegen-units = 1
> panic = "abort"
> strip = "symbols"
> ```

Thanks but I was going for performance, and was only pointing out that it didn't affect the bundle size in a negative way

**Comment by berkus** [comment]
> Thanks but I was going for performance, and was only pointing out that it didn't affect the bundle size in a negative way

Yes, `s` is 2+less inlinling and a lot less loop unrolling, which oftentimes makes code faster by having less impact on caches (depends on the code, of course). `z` disables loop vectorization which may negatively impact perf.

All in all, not bad tradeoff.
